server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java

```diff
@@ -260,6 +261,59 @@ public void testDimension() throws IOException {
         assertThat(e.getCause().getMessage(), containsString("Parameter [dimension] cannot be set"));
     }
 
+    protected <T> void assertMetricType(String metricType, Function<T, Enum<TimeSeriesParams.MetricType>> checker)
```

Comment:
Do you think it'd be worth sticking this in the superclass so we don't have to copy and paste it?